Source/WebCore/history/BackForwardCache.cpp

```diff
@@ -124,7 +124,7 @@ static bool canCacheFrame(LocalFrame& frame, DiagnosticLoggingClient& diagnostic
     }
 
     URL currentURL = documentLoader->url();
-    URL newURL = frameLoader->provisionalDocumentLoader() ? frameLoader->provisionalDocumentLoader()->url() : URL();
+    URL newURL = frameLoader->protectedProvisionalDocumentLoader() ? frameLoader->protectedProvisionalDocumentLoader()->url() : URL();
```

Comment:
You should put provisionalDocumentLoader in a local variable. It's a smell when you have to use the same protected method within the same scope.